arch/power/slide_hash_power8.c

```diff
@@ -10,51 +10,47 @@
 #include "zbuild.h"
 #include "deflate.h"
 
-static inline void slide_hash_power8_loop(deflate_state *s, unsigned n_elems, Pos *table_end) {
+static inline void slide_hash_power8_chain(Pos *table, uint32_t entries, uint16_t wsize) {
     vector unsigned short vw, vm, *vp;
     unsigned chunks;
 
+    table += entries;
+
     /* Each vector register (chunk) corresponds to 128 bits == 8 Posf,
-     * so instead of processing each of the n_elems in the hash table
+     * so instead of processing each of the entries in the hash table
      * individually, we can do it in chunks of 8 with vector instructions.
      *
      * This function is only called from slide_hash_power8(), and both calls
-     * pass n_elems as a power of 2 higher than 2^7, as defined by
-     * deflateInit2_(), so n_elems will always be a multiple of 8. */
-    chunks = n_elems >> 3;
-    Assert(n_elems % 8 == 0, "Weird hash table size!");
+     * pass entries as a power of 2 higher than 2^7, as defined by
+     * deflateInit2_(), so entries will always be a multiple of 8. */
+    chunks = entries >> 3;
+    Assert(entries % 8 == 0, "Weird hash table size!");
 
-    /* This type casting is safe since s->w_size is always <= 64KB
+    /* This type casting is safe since wsize is always <= 64KB
      * as defined by deflateInit2_() and Posf == unsigned short */
```

Comment:
This comment tried to document why casting from `unsigned int` (type of `w_size`) to `Pos` (type of the elements of `head` and `prev`) is fine and safe in this case. Without the cast here (implicitly moved to line 50), the comment doesn't make much sense here anymore.

I don't have a strong opinion if the comment should be moved to line 50 or not.

Reply:
I have removed it.